src/auth/negotiate/UserRequest.cc

```diff
@@ -303,11 +311,17 @@ Auth::Negotiate::UserRequest::HandleReply(void *data, const Helper::Reply &reply
         safe_free(lm_request->server_blob);
         lm_request->request->flags.mustKeepalive = true;
         if (lm_request->request->flags.proxyKeepalive) {
-            const char *tokenNote = reply.notes.findFirst("token");
-            lm_request->server_blob = xstrdup(tokenNote);
+             notes.importIf("negotiate", NoteNames(), [&](const auto &note) {
+                 const auto &name = note->name();
+                 if (!lm_request->server_blob && !name.cmp("token")) {
+                     lm_request->server_blob = xstrndup(name.rawContent(), name.length());
+                     debugs(29, 4, "Need to challenge the client with a server token: '" << name << "'");
+                     return true;
+                 }
+                 return false;
+            });
```

Comment:
While importIf() may improve external ACL code, it is difficult to view the above and similar changes as an improvement. I think we are making this code worse with this approach...

My original idea was to use importIf() _before_ this `switch` statement, to extract all the annotations used by each `case` statement in advance, into local variables, and then use those variables as needed inside each statement. After seeing the above result, I tried to implement that idea myself. It may have improved the overall code look a little, but definitely not enough. Moreover, that implementation lost the ability to detect annotations unused in a specific `case`!

In summary, I do not think we should pursue this importIf() approach, at least not in its current form.